src/main/java/com/google/devtools/build/lib/packages/TargetUtils.java

```diff
@@ -58,6 +58,7 @@ private static boolean legalExecInfoKeys(String tag) {
         || tag.startsWith("disable-")
         || tag.startsWith("cpu:")
         || tag.equals(ExecutionRequirements.LOCAL)
+        || tag.startsWith("timeout")
```

Comment:
This should match either `timeout` exactly, or a `timeout:` prefix (i.e., something like `timeout_blah` shouldn't match).